clippy_lints/src/string_to_string.rs

```diff
@@ -0,0 +1,57 @@
+use crate::utils::is_type_diagnostic_item;
+use crate::utils::span_lint_and_help;
+
+use if_chain::if_chain;
+use rustc_hir as hir;
+use rustc_hir::*;
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+    /// **What it does:** This lint checks for `.to_string()` method calls on values of type `String`.
+    ///
+    /// **Why is this bad?** This does a multitude of things to just clone a string. Using `.clone()` is more specific.
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Example:**
+    ///
+    /// ```rust
+    /// // example code where clippy issues a warning
+    /// let msg = String::from("Hello World");
+    /// let _ = msg.to_string()
+    /// ```
+    /// Use instead:
+    /// ```rust
+    /// // example code which does not raise clippy warning
+    /// let msg = String::from("Hello World");
+    /// let _ = msg.clone()
+    /// ```
+    pub STRING_TO_STRING,
+    restriction,
+    "using `to_string()` on a String, which should be `clone()`"
+}
+
+declare_lint_pass!(StringToString => [STRING_TO_STRING]);
+
+impl LateLintPass<'_> for StringToString {
+    fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &hir::Expr<'_>) {
+        if_chain! {
+            if let ExprKind::MethodCall(path, _, args, _) = &expr.kind;
+            if path.ident.name == sym!(to_string);
+            then {
+                let ty = cx.typeck_results().expr_ty(&args[0]);
+                if is_type_diagnostic_item(cx, ty, sym!(string_type)) {
+                    span_lint_and_help(
+                        cx,
+                        STRING_TO_STRING,
+                        expr.span,
+                        "`to_string()` does a multitude of things to just clone a String",
```

Comment:
Same here, as english is not my mother tongue, my suggestion is maybe good to be ignored :wink: (but keep backticks around `String`)

```suggestion
                        "`to_string()` does a multitude of things than just cloning a `String`",
```